popsycle/synthetic.py

```diff
@@ -3146,3 +3154,454 @@ def calc_f(lambda_eff):
+        if os.path.isfile(output_root2 + '.h5'):
+            raise Exception(
+                'That .h5 file name is taken! Either delete the .h5 file, '
+                'or pick a new name.')
+
+    # Error handling/complaining if input types are not right.
+    if ebf_file[-4:] != '.ebf':
+        raise Exception('ebf_file must be an ebf file.')
+
+    if type(output_root2) != str:
+        raise Exception('output_root must be a string.')
+
+    if type(fdm) != float:
+        if type(fdm) != int:
+            raise Exception('fdm must be a float or an integer.')
+
+    if type(pbh_mass) != int:
+        if type(pbh_mass) != float:
+            raise Exception('pbh_mass must be an integer or a float.')
+
+    if type(r_max) != float:
+        if type(r_max) != int:
+            raise Exception('r_max must be a float or an integer.')
+
+    if type(c) != int:
+        if type(c) != float:
+            raise Exception('c must be an integer or a float.')
+
+    if type(r_vir) != float:
+        if type(r_vir) != int:
+            raise Exception('r_vir must be a float or an integer.')
+
+    if seed is not None:
+        if type(seed) != int:
+            raise Exception('seed must be an integer.')
+
+    # Check to make sure that the output hdf5 file
+    # will not overwrite the input hdf5 file
+    output_hdf5_file = '%s.h5' % output_root2
+    if hdf5_file == output_hdf5_file:
+        raise Exception('Output hdf5 file %s cannot be equal to '
+                        'input hdf5 file %s' % (output_hdf5_file, hdf5_file))
+
+    ##########
+    # Start of code
+    #########
+
+    # Set random seed
+    np.random.seed(seed)
+
+    t0 = time.time()
+
+    #Define parameters for NFW profile calculations
+    r_s = r_vir/c #kpc, scale radius
+    g = 4.3*(10**-3) #(pc*km^2)/(Msun*s^2)
+    h = 70 #km/(s*Mpc)
+
+    #Read in the hdf5 file that doesn't have PBHs. Product of perform_pop_syn.
+    no_pbh_hdf5_file = h5py.File(hdf5_file, 'r')
+    key_list = list(no_pbh_hdf5_file)
+    #Delete lat_bin_edges and long_bin_edges from key_list.
+    key_list = [key for key in key_list if 'bin_edges' not in key]
+
+    #Get data from lat_bin_edges and long_bin_edges
+    lat_bin = no_pbh_hdf5_file['lat_bin_edges'][:]
+    long_bin = no_pbh_hdf5_file['long_bin_edges'][:]
+    bin_edges_number = len(long_bin)
+
+    #Getting the maximum ID from all of the stars and compact objects.
+    #Later used to set the IDs of the PBHs.
+    max_id_no_pbh = []
+    for key in key_list:
+        # max_id_no_pbh.append(np.max(no_pbh_hdf5_file[key][20]))
+        max_id_no_pbh.append(np.max(no_pbh_hdf5_file[key]['obj_id']))
+    max_id = np.amax(max_id_no_pbh)
+
+    hdf5_dset_names = no_pbh_hdf5_file[key_list[0]][:].dtype.names
+
+    no_pbh_hdf5_file.close()
+
+    #Read in ebf file
+    t = ebf.read_ind(ebf_file, '/log', 0)
+    # Convert log to useful dictionary.
+    ebf_log = make_ebf_log(t)
+
+    #Obtain survey area and center latitude and longitude
```

Comment:
Not entirely necessary, but I think it would be helpful in the comment string to note what the units of these three parameters are expected to be (e.g., rad or deg).